ResNet50 trace+2cq Imagenet dataset

mbahnasTT opened the pull request.

punithsekar committed:
#0: yolo evaluation

jbaumanTT committed:
#18058: Move init_sync_registers to TRISC0

Initializing all the sync registers takes around 120 cycles on wormhole. By
moving the work to TRISC0 (when it's already idle) we can cut the cost to BRISC
by around 100 cycles.

nvukobratTT committed:
Extend conv2d TT-Forge sweeps with additional passing and failing cases.

dvartaniansTT committed:
#0: fix a pre commit issue in yolo perf code

dvartaniansTT committed:
#0: fix a pre commit issue in yolo perf code

kmabeeTT committed:
LightMetal - pybind LightMetalReplay, add reset_device fixture to add TTNN Capture+Replay e2e unit tests (Issue #18145)

 - pybind LightMetalReplay class for use in TTNN unit tests
 - rename LightMetalReplay::execute_binary() to LightMetalReplay::run() by suggestion
 - Move test_single_device_light_metal_trace.py to own light_metal subdir which
   now has it's own conftest.py by suggestion.
 - Add reset_device() fixture to tests/ttnn/unit_tests/light_metal/conftest.py
   to close and re-open device between capture + replay
 - Move pair of functions from $ROOT/conftest.py to tests/scripts/common.py
   used by device() and reset_device()
 - Move lightmetal_replay.hpp out of tt_metal/impl into public API folder, which
   prevents TTNN and lightmetal_runner from sneaky includes into tt_metal/impl

nhuang-tt committed:
#0: better test logging

- Log in hex
- Use more bits in the test patterns to ensure
  args are passed without any truncated bits

nhuang-tt committed:
#0: add test case for multiple kernel handles on the same pgm

mbahnasTT committed:
Merge branch 'main' of https://github.com/tenstorrent/tt-metal into main

kmabeeTT committed:
LightMetal - Allow user/test to manage device open/close when using LightMetalReplay (Issue #18145)

 - Makes it easier/possible for TTNN e2e (capture + replay) in pytest
 - Updated tt-metal fixture to use this idea by default, and add unit
   test to cover previous behavior.

nhuang-tt committed:
Generate CRTA multicast commands for all kernels

- Multicast (or unicast for ETH) command was only being created
  for the first kernel. Need to call generate_runtime_arg_cmds for each
  group of CRTAs

nhuang-tt committed:
#18276: add FD test case for CRTAs multiple kernels

TT-BrianLiu committed:
#0: Move ttnn::Shape and compute_strides to tt_metal
- Modify compute_strides logic for handling 0 volume shape

Commit:
Update test cases

mbahnasTT committed:
#0: ResNet50 trace+2cq on Imagenet

mbahnasTT committed:
#0: ResNet50 trace+2cq on Imagenet

mbahnasTT committed:
#0: ResNet50 trace+2cq on Imagenet

mbahnasTT committed:
#0: ResNet50 trace+2cq on Imagenet

sagarwalTT committed:
Add support for page size > max prefetch cmd size for interleaved buffers (#17677)

#16861 

This PR adds support for interleaved buffers to have page sizes which
are greater than the max prefetch command size.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
(https://github.com/tenstorrent/tt-metal/actions/runs/13524331831)
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

umadevimcw committed:
#18495: Update TT_FATAL with blackhole arch check

pavlepopovic commented:
Thanks!
Can we integrate this into models/demos/wormhole/resnet50/demo/demo.py, (and remove the old one) so that is the official demo (and we have only 1 demo script). This is already covered in CI ( https://github.com/tenstorrent/tt-metal/actions/runs/13620607165/job/38069581562). Doesn't make sense to me to have 2 resnet demos.

pavlepopovic commented:
Also when running resnet demo on main at the moment ( `WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest --disable-warnings models/demos/wormhole/resnet50/demo/demo.py::test_demo_sample`), it spits out something like:

`2025-03-03 10:21:42.038 | INFO     | models.demos.ttnn_resnet.demo.demo:run_resnet_inference:205 - inference throughput: 1886.1400786958966 inputs/s`

When running 2cq + trace perf test on main, it spits out this perf:
`2025-03-03 10:22:33.546 | INFO     | models.demos.ttnn_resnet.tests.perf_e2e_resnet50:run_perf_resnet:375 - microsoft/resnet-50 224x224_batchsize16 inference time (avg): 0.003391551971435547, FPS: 4717.604251609819`

Does the new trace + 2cq demo give the perf comparable to 2cq + trace perf test, if not we need to debug where the missmatch is coming?

s-jovic committed:
[skip ci] Use slow matmuls for SD WH pipeline and increase timeout (#18545)

The tests would occasionally fail on the CI; so I enabled `SLOW_MATMULS`
env var for all unit tests to reduce the probability of di/dt occuring.

I also increased the timeout since the tests last a bit longer now.

https://github.com/tenstorrent/tt-metal/actions/runs/13629389522

tt-rkim committed:
#0: Revert "Add support for page size > max prefetch cmd size for interleaved buffers (#17677)"

This reverts commit 07cee7ca1c0ad97526fc99c171dbdce3a50a5d50.

This seems to cause hangs on e2e perf.

omilyutin-tt committed:
Adjust mesh event TTNN APIs  (#18553)

### Ticket
N/A

### Problem description
No need for "mesh_event" suffix, and `wait_for_event` can look exactly
like its single device version.

### Checklist
- [X] New/Existing tests provide coverage for changes

kkabilarTT committed:
Dockerized Profiler Regression (#18516)

### Ticket
Link to Github Issue

### Problem description
Need to update profiler regression tests to run on ubuntu 22.04 -
currently, it is running on 20.04

### What's changed
Containerized the tests to run in a 20.04 environment - where the
container can be called to run on a 22.04 environment

### Checklist

https://github.com/tenstorrent/tt-metal/actions/runs/13608009479

- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: blozano-tt <blozano@tenstorrent.com>
Co-authored-by: Michael Chiou <mchiou@tenstorrent.com>

sminakov-tt committed:
Get device directly from tensor in attn_matmul instead of going through buffer (#18534)

### Ticket

### Problem description
Call to `storage.buffer->device()` will create problems in the future,
when we use MeshDevice to represent single devices.

### What's changed
Replace
`std::get<DeviceStorage>(input_tensor.storage()).buffer->device()->id()`
with simpler `input_tensor.device()`

### Checklist
- [x] [All post commit CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/13613106692)
- [x] New/Existing tests provide coverage for changes

wenbinlyuTT committed:
Fix all_reduce_async ODR violation in unity builds (#18559)

### Ticket
N/A

### Problem description
The pybind function name for `all_reduce_async` clashes with the one for
`all_reduce` in the same namespace (ODR violation). Unity builds are
sensitive to this and could give error like:

```
[1086/1248] Building CXX object ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx.o
FAILED: ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx.o 
/usr/local/bin/ccache /usr/bin/clang++-17 -DFMT_HEADER_ONLY=1 -DTT_ENABLE_LIGHT_METAL_TRACE=1 -I/work/ttnn -I/work/tt_metal -I/work/tt_metal/api -I/work/tt_metal/third_party/umd/device/api -I/work/tt_metal/hostdevcommon/api -I/work/tt_metal/tt_stl -I/work/ttnn/cpp -I/work/build/ttnn/flatbuffers -isystem /work/tt_metal/third_party/tracy/public -isystem /work/.cpmcache/reflect/e75434c4c5f669e4a74e4d84e0a30d7249c1e66f -isystem /work/.cpmcache/xtensor/4a957e26c765b48cbec4a4235fe9e518d5a85d3d/include -isystem /work/.cpmcache/xtensor-blas/190c3a4314355b67291a7d78b20a2100de3f8f54/include -isystem /work/.cpmcache/flatbuffers/2c4062bffa52fa4157b1b4deeae73395df475fda/include -isystem /usr/include/python3.10 -stdlib=libc++ -O3 -std=c++20 -fPIC -fvisibility=default -Werror -Wno-deprecated-declarations -Wdelete-non-virtual-dtor -Wreturn-type -Wswitch -Wuninitialized -Wno-unused-parameter -mavx2 -fPIC -fvisibility-inlines-hidden -fno-lto -Wsometimes-uninitialized -Wno-c++11-narrowing -Wno-error=local-type-template-args -Wno-delete-non-abstract-non-virtual-dtor -Wno-c99-designator -Wno-shift-op-parentheses -Wno-non-c-typedef-for-linkage -Wno-deprecated-this-capture -Wno-deprecated-volatile -Wno-deprecated-builtins -MP -Wno-int-to-pointer-cast -fno-var-tracking -MD -MT ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx.o -MF ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx.o.d -o ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx.o -c /work/build/ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx
In file included from /work/build/ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx:13:
/work/ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce_async/all_reduce_async_pybind.cpp:22:6: error: redefinition of 'bind_all_reduce'
   22 | void bind_all_reduce(pybind11::module& module, const ccl_operation_t& operation, const char* doc) {
      |      ^
/work/ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce/all_reduce_pybind.cpp:22:6: note: previous definition is here
   22 | void bind_all_reduce(pybind11::module& module, const ccl_operation_t& operation, const char* doc) {
      |      ^
In file included from /work/build/ttnn/CMakeFiles/ttnn.dir/Unity/unity_17_cxx.cxx:13:
/work/ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce_async/all_reduce_async_pybind.cpp:103:5: error: no matching function for call to 'bind_all_reduce'
  103 |     detail::bind_all_reduce(
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/work/ttnn/cpp/ttnn/operations/experimental/ccl/all_reduce_async/all_reduce_async_pybind.cpp:22:6: note: candidate template ignored: substitution failure [with ccl_operation_t = registered_operation_t<reflect::fixed_string<char, 37UL - 1>{{{116, 116, 110, 110, 58, 58, 101, 120, 112, 101, ...}}}, ExecuteAllReduceAsync, false>]
   22 | void bind_all_reduce(pybind11::module& module, const ccl_operation_t& operation, const char* doc) {
      |      ^
2 errors generated.
```

### What's changed
Fix the name for `all_reduce_async`

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)

Signed-off-by: wenbinlyuTT <wenbinlyu@tenstorrent.com>

nhuang-tt committed:
Make data movement kernels build at O2 by default (#18356)

### Ticket
N/A

### Problem description
- Currently Compute kernels build at O3 and Data movement kernels build
at Os
- We should make data movement kernels build at at least O2

### What's changed
- Change default for Data movement config to O2
- FD2 and FD tests -- kept at Os
- NOTE: Multicore Fold op kept at Os. Used by resnet

### Checklist
APC

https://github.com/tenstorrent/tt-metal/actions/runs/13540511373/job/37840734705
Single Device Perf
https://github.com/tenstorrent/tt-metal/actions/runs/13584879293
Single Card Demo
https://github.com/tenstorrent/tt-metal/actions/runs/13594610401
T3K
https://github.com/tenstorrent/tt-metal/actions/runs/13594659247
TG/TGG
https://github.com/tenstorrent/tt-metal/actions/runs/13533072675
BH
https://github.com/tenstorrent/tt-metal/actions/runs/13549512017

edwinleeTT committed:
Tightly Pack CB Indices for Reduce + Fused Ops (#18462)

### Ticket
[#16957](https://github.com/tenstorrent/tt-metal/issues/16957)
[#16958](https://github.com/tenstorrent/tt-metal/issues/16958)

### Problem description
CB indices previously had requirements for the values, such as outputs
beginning at 16. These requirements were relaxed but the kernel
implementations remained faithful to them, which hurt dispatch
performance by requiring additional indices to be initialised.

### What's changed
CB indices within the reduction and normalization ops are changed so
that circular buffers use the next available index starting from 0. This
will resolve #16957 and resolve #16958.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI
[passes](https://github.com/tenstorrent/tt-metal/actions/runs/13577752538)
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI [passes
](https://github.com/tenstorrent/tt-metal/actions/runs/13577755233)
- [x] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI
[passes](https://github.com/tenstorrent/tt-metal/actions/runs/13592645262)
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI
[passes](https://github.com/tenstorrent/tt-metal/actions/runs/13592659010)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

tt-aho committed:
#0: Reduce max number of sub-devices to 8

williamlyTT committed:
#17878: [skip ci] Enable failure annotations in models + profiler regression tests (#18552)

### Ticket
#17878

### Problem description
Models and profiler regression tests run inside docker container. 
[pytest-github-actions-annotate-failures requires `GITHUB_ACTIONS=true`
set inside
containers](https://github.com/pytest-dev/pytest-github-actions-annotate-failures?tab=readme-ov-file#usage)

### What's changed
Add `GITHUB_ACTIONS=true` 

### Checklist
- [x] models post commit
https://github.com/tenstorrent/tt-metal/actions/runs/13635241628

nhuang-tt committed:
Delete using namespace in command queue interface header (#18355)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/18325

### Problem description
- `api/command_queue_interface.hpp` has a `using namespace tt::tt_metal`
at the top of the file in the global namespace, causing the entire
repository's global namespace to be polluted and making conflicting
types.

### What's changed
- Delete `using namespace tt::tt_metal` from the header
- Fully qualify types in `tt::tt_metal`in header files.
- Updated the cpp files

### Checklist
Build
https://github.com/tenstorrent/tt-metal/actions/runs/13602340054
On IRD with Ubuntu 22.04: Non unity build and static lib build passing
APC
https://github.com/tenstorrent/tt-metal/actions/runs/13607554703
https://github.com/tenstorrent/tt-metal/actions/runs/13612041035
Blackhole
https://github.com/tenstorrent/tt-metal/actions/runs/13540321674
T3K
https://github.com/tenstorrent/tt-metal/actions/runs/13601542326
Single Card Model Perf
https://github.com/tenstorrent/tt-metal/actions/runs/13610994896
Single Card Demo
https://github.com/tenstorrent/tt-metal/actions/runs/13610993198
Single Card Device Perf
https://github.com/tenstorrent/tt-metal/actions/runs/13610994251

avoraTT committed:
#18430: Increase sem count from 8 to 16 (#18431)

### Ticket
- #18430

### Problem description
Max semaphore count of 8 has been reached, when bringing up a new CCL
op. See #18217 for the new CCL op motivating this increase.

### What's changed
Increase max sem count from 8 to 16.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13636041502)
CI passes
- [x]
[T3k](https://github.com/tenstorrent/tt-metal/actions/runs/13637559960)
CI passes

mbahnasTT committed:
#0: optimizing the trace2cq imagenet demo

mbahnasTT committed:
#0: optimizing the trace2cq imagenet demo

kpaigwar committed:
new function to subtract two corerangesets (#18455)

### Problem description
Added a helper function CoreRangeSet::subtract
This helps to subtract one CoreRangeSet from another.
Basically, it returns A - B = A - (A n B)

### What's changed
* Added a new function called subtract for CoreRangeSet class
* Added pybind for the same
* Added a unit test file
### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13598958290)
CI passes
- [x] New/Existing tests provide coverage for changes

llongTT committed:
Llong/pad wrong output (#18370)

### Ticket
[Link to Github
Issue](https://github.com/tenstorrent/tt-metal/issues/18195)

### Problem description
Padding multi-core kernel run out of memory due to the excessive L1 CB
usage.

### What's changed
Reduce the CB allocation in program factor, ensure large tensor can be
padded using L1 CB.


### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
(https://github.com/tenstorrent/tt-metal/actions/runs/13636872831)
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if
applicable)(https://github.com/tenstorrent/tt-metal/actions/runs/13636870321)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

afuller-TT committed:
Smoke tests (in prep for PR Gate) (#17987)

### Ticket
#17851

### Problem description
Our PR Gate does not run any test at all.

### What's changed
Add Smoke tests that are suitable to run in PR Gate.
Not yet connected to PR Gate as we need to first earmark some runners
for low latency jobs. So that will come as a follow-up.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI
[passes](https://github.com/tenstorrent/tt-metal/actions/runs/13637012956)

yugaoTT committed:
Add missing barrier types for Dynamic NoC (#18486)

Llama model hang when in Dynamic noc mode, the bug was found to be the
interaction with remote cb and dm noc mode.
Bug fixed in this PR also added missing barriers for dm noc

Force-merging because Mo is OOO.

### Checklist
- [x] [All post commit]
https://github.com/tenstorrent/tt-metal/actions/runs/13593269429
- [x] [Blackhole Post commit]
https://github.com/tenstorrent/tt-metal/actions/runs/13594151941
- [x] model perf
https://github.com/tenstorrent/tt-metal/actions/runs/13591271704
- [x] T3K nightly
https://github.com/tenstorrent/tt-metal/actions/runs/13591283467
- [x] T3K model
https://github.com/tenstorrent/tt-metal/actions/runs/13591305402
- [x] T3k unit test
https://github.com/tenstorrent/tt-metal/actions/runs/13594793260
- [ ] TG unit
https://github.com/tenstorrent/tt-metal/actions/runs/13594766139

---------

Co-authored-by: Austin Ho <aho@tenstorrent.com>

rfurko-tt committed:
[tt-train] Enable all GPT-2 like models (#18569)

### Problem description
Unlock all GPT2 models in tt-train. 

### What's changed
* Add custom all_reduce (with moreh_sum)
* Substitute all ttnn::sum with ttnn::moreh_sum before it is fixed
* Add configs for GPT2-(S, M, L, XL)
* Add function to roughly calculate number of parameters 

GPT2-S - 163M
GPT2-M - 406M
GPT2-L - 838M
GPT2-XL - 2.5B (increased number of heads 25->32, embedding dimension
1600 -> 2048; it is done to make TP work)

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13640609836
- [x] New/Existing tests provide coverage for changes

tt-rkim committed:
#0: [skip ci] Properly reference docker image from profiler build for profiler job in BH post commit (#18575)

### Ticket

Quick fix for profiler job in BH

### Problem description

We grabbed the docker image output from `build-artifact` for the
profiler job. Since GHA so graciously fails silently in the case of
typos or invalid references to objects in its embedded JavaScript, we
get an empty `docker-image` input value.

### What's changed

Reference `build-artifact-profiler`

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

nhuang-tt committed:
#0: Remove ThreadManager from api (#18570)

### Ticket
N/A

### Problem description
This code has been here for a while. There's no tests for it and it's
not used anywhere. If users need a ThreadManager they can use something
like boost.

### What's changed
Remove

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

afuller-TT committed:
[skip ci] Fix build when not building tests (#18579)

### Ticket
Follow-up to #17987

### Problem description
When we omit processing the test dirs, CMake fails to configure a target
that assumes test targets exist.

### What's changed
Added a guard to match the guard around the actual tests.

sminakov-tt committed:
Added support for MultiDeviceHostStorage with a single buffer in variety of places (#18532)

### Ticket

### Problem description
When trying to replace Device with MeshDevice, we encounter missing
support for MultiDeviceHostStorage with a single buffer in a variety of
places.

### What's changed
Added handling for MultiDeviceHostStorage in a few places

### Checklist
- [x] [All post commit CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/13613918456)
- [x] New/Existing tests provide coverage for changes

sminakov-tt committed:
Simplify getting a host buffer from tensor to avoid direct calls to command queue (#18533)

### Ticket

### Problem description
Currently we have a few places, where to get a host buffer from tensor
we do some manual logic with calls to command queue, which isn't
supported on MeshDevice.

### What's changed
Move repeating code to a single function and use Tensor::cpu() to
convert device to cpu tensor

### Checklist
- [x] [All post commit CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/13613038021)
- [x] New/Existing tests provide coverage for changes

umadevimcw committed:
#18489: Skip bernoulli test for BH

sminakov-tt committed:
Preserve the same default bottom_up for MeshBuffer (#18587)

### Ticket

### Problem description
Currently Buffer is created using `optional<bool> bottom_up` which
results in different default default values for DRAM and L1 cases.
However, for MeshBuffer `bottom_up` defaults to `false` for all
allocated buffers, which causes unnecessary inconsistency.

### What's changed
Changed `bottom_up` to `std::optional<bool>` in
`DeviceLocalBufferConfig` to preserve the same behavior.

### Checklist
- [x] [All post commit CI
passes](https://github.com/tenstorrent/tt-metal/actions/runs/13648198204)
- [x] New/Existing tests provide coverage for changes

sraizada-tt committed:
Add subdevice support to multicore argmax and plus_one ops  (#16197)

### What's changed
Added sub_core_grids as an arg to use specific cores in the op

### Checklist
- [ ] Post commit CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13629311699
- [x] **(For models and ops writers)** Full [new
models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
tests passes
https://github.com/tenstorrent/tt-metal/actions/runs/12430147704
- [x] New/Existing tests provide coverage for changes

sott0n committed:
Fix missing integer cast for environment variable (#18577)

This PR fixes a type error in the Llama3 demo implementation caused by
loading `MAX_PREFILL_CHUNK_SIZE` as a string instead of an integer.

tt-aho committed:
#0: [skip-ci] Add @tt-aho to fabric codeowners (#18588)

nathan-TT committed:
#0: SFPI update (#18456)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/14597
https://github.com/tenstorrent/sfpi/pull/29
https://github.com/tenstorrent/tt-metal/issues/18340

### Problem description
* New Blackhole feature -- `sfpi::rand()`
* `spushft2` bug fixes

### What's changed
See https://github.com/tenstorrent/sfpi/releases/tag/v6.4.0

ETA: This PR implicitly includes
https://github.com/tenstorrent/tt-metal/pull/18313
(https://github.com/tenstorrent/sfpi/releases/tag/v6.3.0)

### Checklist
- [YES] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [YES] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [YES] New/Existing tests provide coverage for changes

aczajkowskiTT committed:
#13235: topk op fix (#18211)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/13235

### Problem description
few flags & input combinations (k, largest, sorted) were not supported

### What's changed
Original code improperly formatted k in the main function and in the
kernels. The code also contained a lot of repetitive code doing the same
thing. In the solution I fixed aforementioned bugs and refactored the
code into reusable functions.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
tests relevant to ticket, pass but there are a few relevant to cache
size that do not:
https://github.com/tenstorrent/tt-metal/actions/runs/13653198214

This will be resolved in next ticket that references this bug for topk:
https://github.com/tenstorrent/tt-metal/issues/18357

- [x] New/Existing tests provide coverage for changes

jbaumanTT committed:
#0: Add TT_METAL_WORKER_RINGBUFFER_SIZE environment variable

This environment variable is a temporary measure that allows users to set the
worker ringbuffer size to see how that affects performance. It will eventually
be replaced with a device parameter.

Also remove common_runtime_address_map.h, since no kernels are actually using
constants from it, and the values it contains should be retrieved in other
ways.

bbeggsTT committed:
[skip ci] Update llms.md (#18464)

### Ticket
N/A

### Problem description
Order of sections was incorrect.

### What's changed
Section order corrected.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

dchenTT committed:
#18322 Binary NG sharding support uneven shard size (#18324)

### Ticket
[Link to Github
Issue](https://github.com/tenstorrent/tt-metal/issues/18322)

### Problem description
Add support for uneven shard size for HEIGHT sharding. For WIDTH and
BLOCK, still use full shard.

### What's changed
Add more test cases. Add program factory handling. Also add test case
for use_height_and_width_as_shard_shape=False. Many more test cases for
support matrix.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

nardoTT committed:
change directory of tilize_wh kernel (#18580)

### Ticket
Link to Github Issue

### Problem description
The current location of tilize_wh.cpp is causing errors for metal->mlir
uplift

### What's changed
change the location of the kernel to ttnn directory

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13643251562
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

mbahnasTT committed:
#0: Yolov4 post-processing optimization (#18530)

- Leveraging the TM optimization in that PR,
https://github.com/tenstorrent/tt-metal/pull/17790
- Accordingly, consolidating the repeated Eltwise OPs on the small
tensors, by using concat first to merge the small tensors into larger
one. That saved some kernel overall runtime, in addition to reducing the
OP count from ~800 to ~660 (i.e. mainly in the genboxes post-processing
module)
- The reduction in the OP count has positive impact on the trace+2cq e2e
perf.

williamlyTT committed:
#17878: Upload test case data for single card, TG, TGG, L2 pipelines (#18557)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/17878

### Problem description
Single card, TG/TGG, L2 pipelines are missing test case report,
annotation generation + upload

### What's changed
Generate and upload testcase data and annotations for pipelines that are
missing them but are pushing to superset.
Update the single-card and TG workflows callers' list in the
produce_data pipeline to match current workflow names.
Added `uuidgen` to dockerfile (required for test report upload action)

### Checklist (upload step success only)
- [x] (Single-card) Fast dispatch frequent tests
https://github.com/tenstorrent/tt-metal/actions/runs/13636511036
- [x] (Single-card) Device perf regressions
https://github.com/tenstorrent/tt-metal/actions/runs/13643100537
- [x] (Single-card) Model perf tests
https://github.com/tenstorrent/tt-metal/actions/runs/13636551603
- [x] (Single-card) Demo tests
https://github.com/tenstorrent/tt-metal/actions/runs/13636558192
- [x] (TG) TG demo tests
https://github.com/tenstorrent/tt-metal/actions/runs/13638923714
- [x] (TG) TG frequent tests
https://github.com/tenstorrent/tt-metal/actions/runs/13643102605
- [x] (TG) TG model perf tests
https://github.com/tenstorrent/tt-metal/actions/runs/13638933257
- [x] (TG) TG nightly tests
https://github.com/tenstorrent/tt-metal/actions/runs/13643104270
- [x] (TG) TG unit tests
https://github.com/tenstorrent/tt-metal/actions/runs/13639011183
- [x] Nightly tt-metal L2 tests
https://github.com/tenstorrent/tt-metal/actions/runs/13639814502
- [x] (TGG) TGG demo tests
https://github.com/tenstorrent/tt-metal/actions/runs/13643106190
- [x] (TGG) TGG frequent tests
https://github.com/tenstorrent/tt-metal/actions/runs/13643108282
- [ ] (TGG) TGG model perf tests
https://github.com/tenstorrent/tt-metal/actions/runs/13641375089
- [x] (TGG) TGG unit tests
https://github.com/tenstorrent/tt-metal/actions/runs/13643110148

### Checklist
- [x] Rebase + add upload step to PR gate smoke test
- [ ] All post commit
https://github.com/tenstorrent/tt-metal/actions/runs/13656946075

pjanevskiTT committed:
Bump UMD to fix unit tests on P150 (#18567)

### Ticket

Bump UMD to fix failing unit test on 80.15 firmware for Blackhole.
Related to https://github.com/tenstorrent/tt-metal/issues/17178

### Problem description

Test on 80.15 fw was failing because of not reading the harvesting mask
for ETH and DRAM properly.

### What's changed

Harvesting masks are read properly from cluster descriptor now.

### Checklist
- [x] All post-commit tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649803834
- [x] Blackhole post-commit tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13656511777
- [x] (Single-card) Model perf tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649806811
- [x] (Single-card) Device perf regressions :
https://github.com/tenstorrent/tt-metal/actions/runs/13649808945
- [ ] (T3K) T3000 unit tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649811215
- [ ] (T3K) T3000 demo tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649813952
- [ ] (TG) TG unit tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649816216
- [ ] (TG) TG demo tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649817803
- [x] (TGG) TGG unit tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649819292
- [x] (TGG) TGG demo tests :
https://github.com/tenstorrent/tt-metal/actions/runs/13649820697

---------

Co-authored-by: Bojan Rosko <brosko@tenstorrent.com>

wenbinlyuTT committed:
Add per-tensor quantization op to ttnn (#18009)

### Ticket
#15934

### Description
This PR implements simple per-tensor affine
quantize/dequantize/requantize ops for ttnn. The semantics of the API
aligns with the PyTorch equivalent (in terms of how `scale` &
`zero-point` are defined).

Currently the operations convert tensors between bf16 and i32 (instead
of i8/u8) due to limitations in LLK and metal, this will be addressed in
the future. Per-channel quantizaiton is also planned as future work.

Also did some header inclusion cleanups and removing redundant overloads
for `DefaultQueueId`.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Signed-off-by: wenbinlyuTT <wenbinlyu@tenstorrent.com>

akirby-TT committed:
Reverting ac5c168485f498f55374ef9aa1a9ab728af0dac6 (#18628)

### Ticket
Link to Github Issue

### Problem description
Reverting ac5c168485f498f55374ef9aa1a9ab728af0dac6

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

omilyutin-tt committed:
#18572: Add `ttnn.synchronize_device` API for mesh device (#18573)

### Ticket
#18572

### Problem description
Calls that use `ttnn.synchronize_device` and iterate over individual
devices won't work with new TT-distributed abstractions, as we are
abstracting away the synchronization primitives behind the `MeshDevice`.



### What's changed
* Remove `synchronize_devices` helper and instead add a C++ native API
`ttnn.synchronize_device` for both single and mesh device.
* `ttnn.ttnn.synchronize_device` overload for `MeshDevice` will work by
iterating over individual devices, for the time being.
* When the switch to TT-distributed happens, we will replace
`ttnn.synchronize_device` with what `ttnn.synchronize_mesh_device`
currently does, then delete `ttnn.synchronize_mesh_device`.
* This way, we minimize changes to user APIs during the TT-distributed
migration.

### Checklist
- [X] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13641875889)
- [X] [Single card
perf](https://github.com/tenstorrent/tt-metal/actions/runs/13643478606)
- [X] [T3K unit, demo, frequent, nightly, model
perf](https://github.com/tenstorrent/tt-metal/actions/runs/13643793994)
- same failures as on main
- [X] [TG/TGG unit, frequent, model perf,
nightly](https://github.com/tenstorrent/tt-metal/actions/runs/13645062104)
- same failures as on main
- [X] New/Existing tests provide coverage for changes

omilyutin-tt committed:
#18572: Return enqueued events in `ttnn.record_event`, instead of mutating the input argument (#18581)

### Ticket
#18572

### Problem description
Creating "empty" events, and relying on populating them in
`record_event` is error prone, as this intermediate state of an event is
unusable. Moreover, it isn't clear that `record_event` re-populates the
event metadata for the entirely new event.

Current API:
```
op_event = ttnn.create_event(device)
...
# Initialize the op event so we can write
ttnn.record_event(0, op_event)
...
ttnn.wait_for_event(1, op_event)
```

This PR changes this to:
```
op_event = ttnn.record_event(device, 0)
...
ttnn.wait_for_event(1, op_event)
```

### What's changed
* Get rid of `ttnn.create_event` and rely on `ttnn.record_event` to
return events with the populated metadata.

### Checklist
- [X] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13643988015)
- [X] [Single card
perf](https://github.com/tenstorrent/tt-metal/actions/runs/13645074773)
- [X] [T3K unit, demo, frequent, nightly, model
perf](https://github.com/tenstorrent/tt-metal/actions/runs/13645079814)
- same failures as on main
- [X] [TG/TGG unit, frequent, model perf,
nightly](https://github.com/tenstorrent/tt-metal/actions/runs/13645198301)
- same failures as on main
- [X] New/Existing tests provide coverage for changes

daminakaTT committed:
Enable IRAM on ERISC (#18066)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/6740

### Problem description
IRAM should be used to accelerate program execution

### What's changed
- ERISC mac bug workaround
- kernel_init to point to IRAM address
~- TT_METAL_ENABLE_IRAM switch to enable/disable IRAM feature (only on
WH b0)~
- TT_METAL_ENABLE_ERISC_IRAM switch to enable/disable IRAM feature (only
on WH b0)

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

bbradelTT committed:
#18469: use fast_reduce_nc for ttnn.sum front dims (#18603)

### Ticket
Link to Github Issue #18469

### Problem description
For front dimensions, sum transposes first. For some tensors, that step
fails due to size expansion.

### What's changed
- call fast_reduce_nc for sum for front dimensions to avoid the
transpose
- float32 and other generic reduces will be dealt with in a subsequent
PR

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/13663704316
- [x] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
https://github.com/tenstorrent/tt-metal/actions/runs/13663702918
- [x] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
https://github.com/tenstorrent/tt-metal/actions/runs/13657772938
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
https://github.com/tenstorrent/tt-metal/actions/runs/13657770545
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [x] New/Existing tests provide coverage for changes

williamlyTT committed:
#0: Remove --exclude-warning-annotations and suppress junit warning (#18624)

### Ticket
Link to Github Issue

### Problem description
junit warnings thrown by pytest inside profiler regressions show up as
annotations in GHA after
https://github.com/tenstorrent/tt-metal/pull/18106 because the tests run
inside a bash script and don't have `--exclude-warning-annotations` set.
https://github.com/tenstorrent/tt-metal/pull/18229: We don't want to add
`--exclude-warning-annotations` to pytest.ini because that option is
plugin-dependent; we can't assume the env running pytest will have
`pytest-github-actions-annotate-failures` plugin installed

### What's changed
Suppress the junit warning inside pytest.ini (warning is caused by us
adding arbitrary `<properties>` block to testcase xml for CI data
collection purposes and can be safely ignored)
Remove `--exclude-warning-annotations` from workflows. Any
non-junit-related warnings that appear in workflows should not be
suppressed.

### Checklist
- [x] profiler regressions:
https://github.com/tenstorrent/tt-metal/actions/runs/13667617474

sraizada-tt committed:
Fix untilize for sub_core_grids (#18609)

### Ticket
sub_core_grids are not supported for multicore untilize block 

### What's changed
force sub_core_grids multicore untilize to non-blocked version

### Checklist
- [x] [All post
commit][(https://github.com/tenstorrent/tt-metal/actions/runs/13659034287)](https://github.com/tenstorrent/tt-metal/actions/runs/13671716317)
CI passes

mtairum committed:
Remove remaining i2s and s2i BH workaround from to/from_device

dimitri-tenstorrent committed:
#17187: Remove the unnecessary workflow for pushing tt-metal docs to the central site. (#17192)

### Ticket
#17187 

### Problem description

We no longer upload the docs to the central documentation site instead
the docs will be built locally.

### What's changed
We no longer upload the docs to the central documentation site instead
the docs will be built locally.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new
models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
tests passes
- [ ] New/Existing tests provide coverage for changes

sraizada-tt committed:
[skip ci] Remove TG llama demo from CI (#18605)

### Ticket
TG llama demo has ND hangs, and things are being replaced soon.

Need to interface with Models team.

### Problem description
Remove TG llama demo from CI
### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: Raymond Kim <rkim@tenstorrent.com>

tt-rkim committed:
#0: [skip ci] Remove broken llama TG version, dp=4 tp=8 (#18647)

### Ticket

This model is broken and we're replacing it. Removing from README.

### Problem description

This model is broken and we're replacing it.

### What's changed

rm -rf

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

tt-rkim committed:
#0: [skip ci] Remove README models that run on Grayskull because we don't like Grayskull anymore (#18653)

…

### Ticket

Grayskull deprecation

### Problem description

Grayskull deprecation


Remove Grayskull mentions on README

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

sjameelTT committed:
Add fabric init option to conftest (#18521)

### Ticket
#18520

### Problem description
We need to be able to use it via pytest

### What's changed
Add conftest option and a small sanity test that just calls the init for
now

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: Austin Ho <aho@tenstorrent.com>

skrsticTT committed:
Add downsample unit tests (#18644)

### Ticket
Issue: #18201

### What's changed
Downsample tests belonged to tt eager tests, which are
[deleted](https://github.com/tenstorrent/tt-metal/pull/17717). Reverted
same test coverage and put it in unit tests.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes - [my
run](https://github.com/tenstorrent/tt-metal/actions/runs/13678976303)

skrsticTT committed:
Add torch tensor cache to maxpool2d unit tests to speedup test execution (#18649)

The function `torch.randn` takes a significant amount of time while
executing `tests/ttnn/unit_tests/operations/test_maxpool2d.py`. The idea
is to cache torch tensors with specific dimensions and reuse them in
other tests that require tensors of those dimensions. This approach
reduces the test execution time by more than 50% on blackhole.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes - [my
run](https://github.com/tenstorrent/tt-metal/actions/runs/13679665013)

ssinghalTT committed:
refactored

ssinghalTT committed:
refactored

ssinghalTT committed:
refactored

ssinghalTT committed:
refactored

afuller-TT committed:
Afuller/try new ci runners (#18663)

### Ticket
#18664

### Problem description
We want to start using the new CI runners. But first we want to test
them out a bit.

### What's changed
Scheduled a job outside of APC to run on a new CI runner.
The same task when run inside APC will continue to run on the existing
runners to minimize potential disruption.
Checked the syntax by having a run that targeted each type:
https://github.com/tenstorrent/tt-metal/actions/runs/13666458620

tt-rkim committed:
#0: [skip ci] Ensure we don't hang on upload in case of network issues by using a timeout, as the upload should complete for the device perf files pretty quickly (#18665)

### Ticket

Quick fix.

### Problem description

Upload could hang on network issues, but nothing actually timed it out
so it could hang the runner.

### What's changed

Add a timeout on upload step to ensure we exit, as the upload should
always work and be consistently fast.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

williamlyTT committed:
#0: [skip ci] Add 10 minute timeout to all download + upload artifact steps (#18667)

### Ticket
Link to Github Issue

### Problem description
Download and upload can hang on network issues, without timeouts we
stall the job queue.

### What's changed
Add 10 min timeouts for:
-  `actions/download-artifact`
-  `actions/upload-artifact-with-job-uuid`
-  `actions/upload-artifact`

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

abhullar-tt committed:
Bumping UMD to pick up fix for eth translation on p150 (#18654)

### Ticket
N/A 

### Problem description
Fix for eth physical to translated coordinates checked into umd main,
need to pick it up to unblock p150 dev/testing


### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13681038365)
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/runs/13681046728)

mbahnasTT commented:
@pavlepopovic short answer is the 4700 fps is going down to ~90 fps!, and I did a breakdown analysis and filed an issue:
https://github.com/tenstorrent/tt-metal/issues/18607 and we added a 'TODO` comment in the code to acknowledge the point.
The breakdown:
- The `ttnn.from_torch` and `ttnn.from_device` before/after the execute trace impacted the runtime to be 2000 fps
- The `get_batch()` dropped it to 90 fps

We need to implement a fast input data streamer and investigate on the slow ttnn commands I mentioned.

tt-aho reviewed (approved):
One of the main reasons for slowdown here compared to the perf test is that we are issuing blocking reads here, and we aren't enqueuing the next write + forward pass while the current forward pass is running, so there isn't much benefit being had here using 2cqs. Things are basically serialized.

mbahnasTT commented on models/demos/ttnn_resnet/tests/resnet50_performant_imagenet.py:
```diff
@@ -0,0 +1,116 @@
+        # TODO: Add in place support to ttnn to_memory_config
+        self.input_tensor = ttnn.reshard(self.tt_image_res, self.input_mem_config, self.input_tensor)
+        ttnn.record_event(0, self.op_event)
+        ttnn.execute_trace(self.device, self.tid, cq_id=0, blocking=True)
+        outputs = ttnn.from_device(self.test_infra.output_tensor, blocking=True)
```
OK, I can change it on the execute_trace to False, but I'd need to keep it true on the following line, because it's issuing an error

tt-aho replied:
yes. With the current setup you need to have the read be blocking. There are ways to refactor the code to be able to do non-blocking reads here so that we can enqueue more work ahead of time before we actually wait for the output.

ssinghalTT replied:
@tt-aho , do you have an example I can look at to implement the steps that its non-blocking?

ssinghalTT committed:
updated the event code to latest, non blocking on execute_trace

ssinghalTT committed:
Merge branch 'mbahnas/rn50_trace2cq_imagenet' of https://github.com/tenstorrent/tt-metal into mbahnas/rn50_trace2cq_imagenet

pavlepopovic commented:
@sagarwalTT can you clean up the branch so it doesn't have 84 commits and 800 changed files pls? (do a rebase on main with relevant commits only)

ssinghalTT commented:
@pavlepopovic , yes created a new branch and PR https://github.com/tenstorrent/tt-metal/pull/18732
